src/main/java/org/dataloader/DataLoader.java

```diff
@@ -562,10 +560,12 @@ public int dispatchDepth() {
      * @param error the exception to prime instead of a value
      * @return the data loader for fluent coding
      */
-    public DataLoader<K, V> prime(K key, Exception error) {
+    public DataLoader<K, V> prime(K key, Throwable error) {
         Object cacheKey = getCacheKey(key);
-        if (!futureCache.containsKey(cacheKey)) {
-            futureCache.set(cacheKey, CompletableFutureKit.failedFuture(error));
+        synchronized (this) {
+            if (!valueCache.containsKey(cacheKey)) {
+                valueCache.set(cacheKey, error);
+            }
```

Comment:
The more I look at this, the more I'm not sure if it's right. Initially, I changed the `CacheMap#get` function to return a `Try` to allow priming the cache with an error, but I'm wondering if that is right. Maybe we just keep an independent cache of errors to support priming failures, but not let the CacheMap worry about that. That makes a lot more sense to me than caching errors, which are often times much more temporary than a cache would be.